Hardware/Rpi4.md

```diff
@@ -0,0 +1,97 @@
+---
+arm_hardware: true
+cmake_plat: rpi4
+xcompiler_arg: -DAARCH64=1
+platform: Raspberry Pi 4-b
+arch: ARMv8A
+virtualization: ARM HYP
+iommu: "No"
+soc: BCM2711
+cpu: Cortex-A72
+Status: Unverified
+Contrib: "[Hensoldt](https://hensoldt-cyber.com) and [ARM Research IceCap](https://gitlab.com/arm-research/security/icecap/icecap)"
+Maintained: "[Hensoldt](https://hensoldt-cyber.com)"
+SPDX-License-Identifier: CC-BY-SA-4.0
+SPDX-FileCopyrightText: 2020 seL4 Project a Series of LF Projects, LLC.
+---
+
+# Raspberry PI 4 Model B
+
+## Serial connection
+Serial TX and RX are located at GPIO pins 14 and 15 respectively
+
+## U-Boot
+
+Right now, the default U-Boot will not successfully boot an seL4 image
+on the RPi4 because of cache configuration issues in the seL4
+ELFLoader. This problem can be remedied by having U-boot disable caches
+before loading seL4. Unfortunately, the stock upstream U-boot used to
+disable caches before loading the kernel image, but as of this patch
+(<https://github.com/u-boot/u-boot/commit/995eab8b5b580b67394312b1621c60a71042cd18>),
+U-boot no longer disables caches.
+
+In order to obtain a U-boot binary that disables caching, you should
+compile U-Boot from source yourself.
+You can do this by cloning U-boot from upstream,
+then reverting commit 995eab8b5b580b67394312b1621c60a71042cd18, and then
+building it:
+```bash
+git clone https://github.com/u-boot/u-boot.git u-boot
+cd u-boot
+git revert 995eab8b5b580b67394312b1621c60a71042cd18
+make CROSS_COMPILE=aarch64-linux-gnu- rpi_4_defconfig
+make CROSS_COMPILE=aarch64-linux-gnu-
+```
+
+This will enable you to get the most up-to-date U-boot which will boot
+seL4 on the RPi4 successfully.
+
+**NOTE:** Automatic revert is not going to work if you are using the
+latest version of u-boot. You need to manually revert the change by
+looking at the changeset.
+
+
+## SD card setup
+ The PI boots from the first FAT32 partition on the
+SD card. Where files are specified, they should be located in the root
+directory of this partition.
+
+|Stage |Filename |Description |Source|
+|-|-|-|-|
+|FSBL |- |Mounts SD and loads SSBL |ROM |
+|SSBL |bootcode.bin|Loads GPU firmware and boots GPU|<https://github.com/raspberrypi/firmware/tree/master/boot> |
+|GPU firmware |start4.elf |Loads CPU bootloader and boots CPU |<https://github.com/raspberrypi/firmware/tree/master/boot> |
+|Usually the Linux kernel, but could also be u-boot |u-boot.bin |u-boot| Compiled using the instructions above |
+||config.txt|u-boot parameters |Add `arm_64bit=1`, `kernel=u-boot.bin`, and `dtoverlay=pi3-disable-bt` to the bottom of config.txt |
```

Comment:
At https://github.com/raspberrypi/firmware/blob/master/boot/overlays/README is says that `pi3-disable-bt` has been renamed `disable-bt`, but the old name is kept as alias. To avoid confusion why RasPi3 things are used here, this should use the new name.